src/update_notifier.rs

```diff
@@ -102,16 +102,10 @@ impl WapmUpdate {
                 let old_version = util::get_latest_runtime_version("wasmer")?;
 
                 if !force_update_notification {
-                    if let Some(b) = util::compare_versions(&old_version, &new_version) {
-                        if b {
-                            return Ok(());
-                        }
-                    } else {
-                        // fall back to direct comparison
-                        // If we are in the same version
-                        if old_version == new_version {
-                            return Ok(());
-                        }
+                    let compare = util::compare_versions(&old_version, &new_version);
+                    match compare {
+                        Ok(crate::util::VersionComparison::NewIsGreater) => {}
```

Comment:
does this even compile?
It doesn't seem to return anything.

Reply:
If `NewIsGreater`, then it will pass (and print the message).

If not `NewIsGreater`, then it will just return.

It could possibly be just `return;` instead, because printing the message isn't supposed to crash wapm completely.